src/main/java/com/conveyal/datatools/manager/DataSanitizer.java

```diff
@@ -0,0 +1,237 @@
+package com.conveyal.datatools.manager;
+
+import com.conveyal.datatools.common.utils.aws.CheckedAWSException;
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import com.conveyal.gtfs.GTFS;
+import com.conveyal.gtfs.util.InvalidNamespaceException;
+import com.conveyal.gtfs.util.Util;
+import com.google.common.collect.Lists;
+import com.mongodb.client.model.Projections;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.conveyal.datatools.manager.DataManager.GTFS_DATA_SOURCE;
+import static com.conveyal.datatools.manager.DataManager.initializeApplication;
+import static com.mongodb.client.model.Aggregates.project;
+import static com.mongodb.client.model.Filters.nin;
+
+/**
+ * The Data sanitizer requires the env.yml and server.yml files for configuration. Data sanitizer specific parameters
+ * should be provided after these e.g.:
+ * configurations/test/env.yml.tmp configurations/test/server.yml.tmp --orphaned --delete
+ */
+public class DataSanitizer {
+    private static final Logger LOG = LoggerFactory.getLogger(DataSanitizer.class);
+    public static final List<String> ORPHANED_FLAGS = List.of("--orphaned", "-O");
+    public static final List<String> DELETE_FLAGS = List.of("--delete", "-D");
+
+    public static void main(String[] args) throws IOException {
+        initializeApplication(args, false);
+        parseArguments(args);
+    }
+
+    /**
+     * Parse the arguments provided on the command line. The first two arguments must reference the env.yml and
+     * server.yml files.
+     */
+    public static void parseArguments(String[] arguments) {
+        Map<String, Set<String>> groupedParams = parseCommandLineArguments(arguments);
+        groupedParams.forEach((flag, commandValues) -> {
+            if (ORPHANED_FLAGS.contains(flag)) {
+                sanitizeFeedVersions(DELETE_FLAGS.contains(flag));
```

Comment:
If `flag` is `--orphaned` or `-O`, it can never be also `--delete` or `-D`, so orphaned feed versions will never be deleted as written.

Reply:
Doh! I have change the delete items to commands instead of flags.